lock-api/src/main/java/com/palantir/lock/LockClient.java

```diff
@@ -66,6 +68,11 @@ public static LockClient of(String clientId) {
         Preconditions.checkArgument(!clientId.equals(INTERNAL_LOCK_GRANT_CLIENT_ID));
         return new LockClient(clientId);
     }
+    
+    @JsonCreator
+    public static LockClient valueOf(String clientId) {
```

Comment:
Any reason not to annotate `LockClient::of(String)` as `@JsonCreator` instead of another factory?